src/expr/src/scalar/func.rs

```diff
@@ -6643,11 +6717,35 @@ impl VariadicFunc {
     /// introduces nulls even when it does not.
     pub fn introduces_nulls(&self) -> bool {
         use VariadicFunc::*;
-        // Note the negation
-        !matches!(
-            self,
-            And | Or //todo: add more
-        )
+        match self {
+            Concat
+            | PadLeading
+            | Substr
+            | ArrayCreate { .. }
+            | ListCreate { .. }
+            | RecordCreate { .. }
+            | ListSliceLinear
+            | SplitPart
+            | HmacString
+            | HmacBytes
+            | ErrorIfNull
+            | DateBinTimestamp
+            | DateBinTimestampTz
+            | RangeCreate { .. }
+            | And
+            | Or => false,
+            Coalesce
+            | Greatest
+            | Least
+            | MakeTimestamp
+            | Replace
+            | JsonbBuildArray
+            | JsonbBuildObject
```

Comment:
It seems to me that `JsonbBuildObject` and `JsonbBuildArray` return null only if one of the inputs are null.

Reply:
You're right; I misinterpreted the slice of `Datum`s.